src/components/footer/join-newsletter/join-newsletter.tsx

```diff
@@ -0,0 +1,49 @@
+import { useState } from "react";
+import { SingleInputField } from "../single-input-field/single-input-field";
+import styles from "./join-newsletter.module.scss";
+
+export const JoinNewsletter = () => {
+  const [emailSent, setEmailSent] = useState(false);
+  const [serverError, setServerError] = useState(false);
+
+  const onHandleSubmit = async (value: string) => {
+    try {
+      const response = await fetch("http://localhost:3000/marcin", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({ email: value }),
+      });
+
+      if (response.ok) {
+        setEmailSent(true);
+      } else {
+        setServerError(true);
+      }
+    } catch (error) {
+      console.error("Error sending email:", error);
```

Comment:
we are still displaying console log here, why

Reply:
if we catch error we should display to the user Something went wrong, just the same if the response is not okay

Reply:
so instead of console.error , setServerError(true)